### PR TITLE
[ENH] making reported run distance match route distance

### PR DESCRIPTION
**Pull request by rksh**
making the distance calc conditional match route addition - distances will match more closely.
TODO: eliminate low-accuracy jitter (still a problem)

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java [hunk 1]
```diff
@@ -74,8 +74,6 @@ public final class ListFragment extends Fragment implements ApiListener, DialogL
     private static final int UPLOAD_DIALOG = 101;
     private static final int QUICK_PAUSE_DIALOG = 102;
 
-    public static final float MIN_DISTANCE_ACCURACY = 32f;
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:77]: removing in favor of centralized MIN_ROUTE_LOCATION_PRECISION_METERS

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java [hunk 2]
```diff
@@ -183,7 +182,7 @@ public void onCreate(final Bundle savedInstanceState) {
             locale = Locale.US;
         }
         numberFormat = NumberFormat.getNumberInstance(locale);
-        numberFormat.setMaximumFractionDigits(1);
+        numberFormat.setMaximumFractionDigits(2);
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java:185]: adding two decimals facilitates debugging as well as helping with the map view.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java [hunk 3]
```diff
@@ -27,13 +27,14 @@
...
 import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicLong;
 
 import androidx.core.content.ContextCompat;
 
 public class GPSListener implements Listener, LocationListener {
     public static final long GPS_TIMEOUT_DEFAULT = 15000L;
     public static final long NET_LOC_TIMEOUT_DEFAULT = 60000L;
-    public static final float LERP_MIN_THRESHOLD_METERS = 1.0f;
+    public static final float LERP_MIN_THRESHOLD_METERS = 20.0f;
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:37]: am i nuts, or should we avoid linear interpolation unless the difference is a little larger than 1m

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java [hunk 4]
```diff
@@ -254,35 +257,41 @@ else if ( newOK && NETWORK_PROVIDER.equals( newLocation.getProvider() ) ) {
...
-                            dist + prefs.getFloat( ListFragment.PREF_DISTANCE_RUN, 0f ) );
-                    edit.putFloat( ListFragment.PREF_DISTANCE_TOTAL,
-                            dist + prefs.getFloat( ListFragment.PREF_DISTANCE_TOTAL, 0f ) );
-                    edit.apply();
-                }
-                if ( dist > LERP_MIN_THRESHOLD_METERS && dbHelper != null) {
-                    if (dist > LERP_MAX_THRESHOLD_METERS) {
-                        Logging.warn("Diff is too large, not lerping. " + dist + " meters");
-                        dbHelper.clearPendingObservations();
+        if (location != null && location.getTime() != 0L &&
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:260]: See DatabaseHelper.logRouteLocation (:1163) for almost-exactly-the-same-test

Comment (bobzilladev) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:260]: This removes the check for GPS_PROVIDER, is that intentional?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java (hunk 3 above)
Comment (uhtu) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:37]: why choose?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java (hunk 4 above)
Comment (uhtu) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:260]: is that the source of the difference?

**Review by uhtu — APPROVED** [review]
anyhow, this should at least align the two.
👍

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java (hunk 4 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:260]: _A_ source- but not the only one. You've seen the routes on your various devices, they don't look terrible is most cases.

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:260]: There's probably a second PR where I try to eliminate jitter in both distance and route while holding still.

**Review by bobzilladev — APPROVED** [review]

**Pull request merged**